conda/install.py

```diff
@@ -922,6 +927,8 @@ def move_path_to_trash(path):
         except OSError as e:
             log.debug("Could not move %s to %s (%s)" % (path, trash_dir, e))
         else:
+            if path in linked_data_:
+                del linked_data_[path]
```

Comment:
I made a change for this, but when trying to test it, I rebased then ran into an issue where MatchSpec.strictness is no more since https://github.com/conda/conda/commit/6ddc6d9845a49f25fa67ed6702c409168b2be1e8, but conda-build still refers to it.

I guess:

```
    spec.strictness == 1 => spec.is_simple()
    spec.strictness == 3 => spec.is_exact() 
```

Please review https://github.com/conda/conda-build/pull/959